src/newrelic/utilization/utilization_hash.go

```diff
@@ -204,6 +199,38 @@ func GatherDockerID(util *Data) error {
 	return nil
 }
 
+func OverrideDockerId(util *Data, id string) {
+	if nil == util.Vendors {
+		util.Vendors = &vendors{}
+		util.Vendors.Docker = &docker{}
+	}
+	util.Vendors.Docker = &docker{ID: id}		
+}
+
+func OverrideVendors(util *Data) {
+	if util.Vendors.isEmpty() {
+		// Per spec, we MUST NOT send any vendors hash if it's empty.
+		util.Vendors = nil
+	}
+}
+
+func GetDockerId(util *Data) (string, error) {
+    id := ""
+    if nil == util {
+        return id, fmt.Errorf("Util is nil")
+    }
+    if util.Vendors.isEmpty() {
```

Comment:
Do you need to check that `util.Vendors` is not `nil`?  Above when you create these structs you allocation `util.Vendors` separately from the `util` struct.

Reply:
I believe the logic to check if `Vendors == nil` is handled implicitly by the call to the isEmpty function here: https://github.com/newrelic/newrelic-php-agent/blob/main/src/newrelic/utilization/utilization_hash.go#L70

Reply:
OK this should work with how golang handles data types and functions for data types.